tests/osaka/eip7823_modexp_upper_bounds/test_modexp_upper_bounds.py

```diff
@@ -0,0 +1,128 @@
+"""
+abstract: Test [EIP-7823: Set upper bounds for MODEXP](https://eips.ethereum.org/EIPS/eip-7823)
+    Tests upper bounds of the MODEXP precompile.
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork, Osaka
+from ethereum_test_tools import Account, Alloc, Environment, StateTestFiller, Transaction
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from ...byzantium.eip198_modexp_precompile.helpers import ModExpInput, ModExpOutput
+from ..eip7883_modexp_gas_increase.spec import Spec, Spec7883
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7823.md"
+REFERENCE_SPEC_VERSION = "c8321494fdfbfda52ad46c3515a7ca5dc86b857c"
+
+MAX_LENGTH_BYTES = 1024
+
+
+@pytest.fixture
+def precompile_gas(fork: Fork, mod_exp_input: ModExpInput) -> int:
+    """Calculate gas cost for the ModExp precompile and verify it matches expected gas."""
+    spec = Spec if fork < Osaka else Spec7883
+    calculated_gas = spec.calculate_gas_cost(
+        len(mod_exp_input.base),
+        len(mod_exp_input.modulus),
+        len(mod_exp_input.exponent),
+        mod_exp_input.exponent_value,
+    )
+    return calculated_gas
+
+
+@pytest.mark.valid_from("Prague")
+@pytest.mark.parametrize(
+    "mod_exp_input",
+    [
+        pytest.param(
+            ModExpInput(
+                base=b"\0" * MAX_LENGTH_BYTES,
+                exponent=b"\0" * MAX_LENGTH_BYTES,
+                modulus=b"\0" * (MAX_LENGTH_BYTES - 1) + b"\2",
+            ),
+            id="max_length",
+        ),
+        pytest.param(
+            ModExpInput(
+                base=b"\0" * (MAX_LENGTH_BYTES + 1),
+                exponent=b"\0" * MAX_LENGTH_BYTES,
+                modulus=b"\0" * (MAX_LENGTH_BYTES - 1) + b"\2",
+            ),
+            id="excess_length_base",
+        ),
+        pytest.param(
+            ModExpInput(
+                base=b"\0" * (MAX_LENGTH_BYTES),
+                exponent=b"\0" * (MAX_LENGTH_BYTES + 1),
+                modulus=b"\0" * (MAX_LENGTH_BYTES - 1) + b"\2",
+            ),
+            id="excess_length_exponent",
+        ),
+        pytest.param(
+            ModExpInput(
+                base=b"\0" * (MAX_LENGTH_BYTES),
+                exponent=b"\0" * (MAX_LENGTH_BYTES),
+                modulus=b"\0" * (MAX_LENGTH_BYTES) + b"\2",
+            ),
+            id="excess_length_modulus",
+        ),
```

Comment:
```suggestion
        pytest.param(
            ModExpInput(
                base=b"\0" * (MAX_LENGTH_BYTES + 1),
                exponent=b"\0",
                modulus=b"\2",
            ),
            id="excess_length_base",
        ),
        pytest.param(
            ModExpInput(
                base=b"\0",
                exponent=b"\0" * (MAX_LENGTH_BYTES + 1),
                modulus=b"\2",
            ),
            id="excess_length_exponent",
        ),
        pytest.param(
            ModExpInput(
                base=b"\0",
                exponent=b"\0",
                modulus=b"\0" * (MAX_LENGTH_BYTES) + b"\2",
            ),
            id="excess_length_modulus",
        ),
```

Had to rework this for it to be compatible with EIP-7825, but it works now 👍